docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+from the search API like any other field. The <<search-field-caps,field capabilities API>> returns runtime fields as both searchable and aggregatable.
+
+By dynamically evaluating runtime fields at search time, you can optimize disk
+space by choosing which fields to index. If you later decide that a runtime
+field is increasingly used for filtering or aggregation, you can add the field
+to `_source` to gain faster search speed.
+
+Alternatively, you can specify a `runtime_mappings` section in a search request
+to use runtime fields that exist only as part of the query. This
+flexibility allows you to create or override field values in `_source` for the
+duration of the query without modifying the field itself.
+
+Runtime fields help to alleviate several common issues when using {es}:
+
+* Reindexing your data between development iterations is slow and can make
+experimenting on large datasets difficult
+* Indexing data before searching makes running one-off searches costly and
+resource intensive
+* Indexing all of your data instead of just the fields you want to search
+requires more disk space to gain performance at search time
+* Reindexing data for time-based indices to ensure that existing indices
+include any new fields in the index template is slow
+* Determining how a field is evaluated in {kib} at index or search time is
+difficult for scripted fields because they have different needs based on where
+they are implemented
+
+Runtime fields incur performance costs at search time, depending
+on the <<runtime-params-runtime-type,`runtime_type`>>. For example, let's say
+you created an anomaly detection job that operates on the `@timestamp` field.
+If the `@timestamp` field is a runtime field, the search cost would be extremely
+high because the data isn't indexed and {es} must compute the value for each
+document that matches the query.
+
+NOTE: Computing values for runtime fields in each document that might match a
+query impacts search speed. Use the <<async-search,asynchronous search API>>
+to run searches that include runtime fields.
+
+Runtime fields are useful when working with log data, especially when you're
+unsure about the data structure. Your search speed decreases, but your index
+size is much smaller and you can more quickly process logs without having to
+index them.
+
+[[runtime-mapping-fields]]
+==== Mapping a runtime field
+You map runtime fields by adding a `"runtime"` section under the mapping
+definition. Within that section, you
+<<modules-scripting-using,define a Painless script>>, which has access to the
+entire context of a document, the original `_source`, and the mapped field plus
+its values (`doc_values`). At search time, the script runs and generates values
+for each scripted field.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request. You can retrieve the results of running a script, but
+also make the script results available for queries and aggregations.
```

Comment:
I think this might be missing a conjunction for it to make sense to me - runtime field are similar to `script_fields` *but* **words words words**.

Reply:
I think we can just shorten to:
>Runtime fields are similar to the script_fields parameter of the _search request, but also make the script results available for queries and aggregations.

Reply:
:+1: